public/assets/js/impact_discover/impact_discover.js

```diff
@@ -0,0 +1,261 @@
+    resetODSIcons();
+    resetODScheckboxes();
+    var odsActive = getActiveODS();
+
+    var odsArray = [];
+
+    $.each(odsActive, function(key,ods){
+        addODSIcon(ods);
+        checkODSoption(ods);
+        odsArray.push(ods.id);
+    });
+
+    resetProjects();
+    loadProjects(odsArray);
+}
+
+function resetProjects() {
+    $('.impact-discover-projects > div.container > div:not(#ods-icons)').remove();
+}
+
+function loadProjects(ods) {
+    $('.impact-discover-projects > div.container').after('<div class="loading-container">Loading</div>');
+    var url = "https://goteo-seven.vercel.app/ods.html?ods=";
+    var url = url+ods.join(",");
+    $.get( url, function( data ) {
+        $('.impact-discover-projects > div.container').append( data );
+      })
+      .done(function(){
+        $('.loading-container').remove();
+      });
+}
+
+// reset footprints active status
+function resetFootprints() {
+    $("a[data-footprint]").removeClass("active");
+}
+
+// activate footprints by ods selected
+function activateFootprints(footprints) {
+    $.each(footprints, function(key, footprint){
+        $('a[data-footprint="'+footprint+'"]').addClass("active");
+    })
+}
+
+// filter ODS select options by footprint
+function filterODSByFootprint (footprint) {
+    resetODSSelect();
+    resetODS();
+    var options = odsList.ods.filter(function(ods) {
+        return hasFootprint(ods,footprint) ? ods.ods : false ;
+    });
+    $.each(options,function(key,option){
+        if (option) {
+            activateODS(option);
+        }
+    });
+    refreshODS();
+}
+
+// filter footprints by ODS
+function filterFootprintByODS (ods) {
+    resetFootprints();
+    resetODSIcons();
+    activateFootprints(ods);
+}
+
+function expandMobileFilter() {
+    $("#filters-footprints").slideDown();
+    $("#filters-ods-list").slideDown();
+    $("#filters-mobile").css({"width":"100%"});
+    $("#filters-mobile .close").show();
+}
+
+function contractMobileFilter() {
+    $("#filters-footprints").slideUp();
+    $("#filters-ods-list").slideUp();
+    $("#filters-mobile .close").hide();
+    $("#filters-mobile").css({"width":"50%"});
+}
+
+function needsScroll() {
+    var $last = $('.impact-discover-projects div:last');
+    return $(window).scrollTop() >= $last.offset().top - $last.height()
+}
+
+$(window).on('resize scroll', function() {
+    if(needsScroll() && !$('.loading-container').length) {
+        ods = [];
+        //TODO: find selected ODS
+        var odsActive = getActiveODS();
+
+        var odsArray = [];
+
+        $.each(odsActive, function(key,ods){
+            odsArray.push(ods.id);
+        });
+        loadProjects(odsArray);
+    };
+});
+
+// bind click on footprint
+$("#filters-footprints").on("click","a", function(e){
+    e.preventDefault();
+    resetFootprints();
+    footprint = $(this).attr("data-footprint");
+    filterODSByFootprint(footprint);
+    $(this).addClass("active");
+    if ($(window).width()<721) {
```

Comment:
@davidbeig why 721? Is this a value that could be set as a constant in a class so we can reuse it across the whole project?

Reply:
True, this value was chosen by Jordi, inside our scss you can find the definitions we have for the different breakpoints inside _variables.scss. In this case we define it like this

```
$breakpoint-sm: 768px;
```

We should really import this defined variables to our javascript